File: 08-Testing-Java/Starter/test/testing/tests/ArchitectureTests.java
package testing.tests;

import com.tngtech.archunit.core.domain.JavaClasses;
import com.tngtech.archunit.core.importer.ClassFileImporter;
import com.tngtech.archunit.lang.ArchRule;
import org.junit.jupiter.api.Test;

import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
import static org.junit.jupiter.api.Assertions.fail;

public class ArchitectureTests {
    @Test
    public void typesInProductiveCode_MustNotDirectlyAccessSystemOut() {
        fail("Exercise 2: Verify that types in productive code do not access System.out");
    }

    @Test
    public void testDoublesInTestCode_MustHavePrefixFakeInClassName() {
        fail("Exercise 2: Verify that test doubles defined in test code have the prefix Fake in the class name");
    }
}

File: 08-Testing-CSharp/Starter/Testing.Tests/ArchitectureTests.cs
namespace Testing.Tests;

using ArchUnitNET.Domain;
using ArchUnitNET.Loader;
using ArchUnitNET.Fluent;
using ArchUnitNET.xUnit;

using static ArchUnitNET.Fluent.ArchRuleDefinition;

public class ArchitectureTests
{
    [Fact]
    public void TypesInProductiveCode_MustNotDirectlyAccessSystemConsole()
    {
        Assert.Fail("Exercise 2: Verify that types in productive code do not access Console.WriteLine()");
    }

    [Fact]
    public void TestDoublesInTestCode_MustHavePrefixFakeInClassName()
    {
        Assert.Fail("Exercise 2: Verify that test doubles defined in test code have the prefix Fake in the class name");
    }
}